QuickOSM/ui/base_processing_panel.py

```diff
@@ -55,6 +56,11 @@ def setup_panel(self):
             QgsFileWidget.GetDirectory)
         self.dialog.output_directories[self.panel].setDialogTitle(tr('Select a directory'))
 
+        self.dialog.output_format[self.panel].addItem('GeoJSON', Format.GeoJSON)
```

Comment:
You can store extra attributes in the enum, using namedtuple. One for "label" for instance.